x509/golang.go

```diff
@@ -52,7 +53,11 @@ func genCertificate(
 }
 
 func parsePemCertificateRequest(csrPem string) *x509.CertificateRequest {
-	pemBlock, _ := pem.Decode([]byte(csrPem))
+	pemBlock, rest := pem.Decode([]byte(csrPem))
+	if pemBlock == nil || len(rest) > 0 {
+		subcommands.DieNotNil(errors.New("Malformed PEM data"))
+		return nil
```

Comment:
Do we need to return after "dying"?

Why are we "dying" inside of some low-level function, maybe it makes sense to return nil/err and die within the caller context?

Reply:
The `return` is because of the linter.

The dying is to not change the interface and all associated function calls.
We do `DieNotNil` in many low-level functions across Fioctl; I see no problem with that as long as it helps keep the code more concise.

Reply:
>We do DieNotNil in many low-level functions across Fioctl; I see no problem with that as long as it helps keep the code more concise.

Fair enough, I suppose that's fine as long as there is no need to do some specific cleanup upper in the stack that is not automatically invoked when the "die" is called.

Reply:
hmm... interesting and actually valid point.

But, we do no cleanups ATM... so I will leave it for another day.